src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1238,7 +1233,7 @@ private Action<Diagnostic> GetAddDiagnostic(ISymbol contextSymbol, ImmutableArra
         {
             return diagnostic =>
             {
-                if (isGeneratedCodeSymbol && shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
+                if (shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
```

Comment:
by looking at the code, cost would be proportional to d (number of diagnostics reported) * t (number of trees in the compilation). so, if there is a lot of diagnostics in a big solution, cost seems can be quite big.